include/zephyr/drivers/emul.h

```diff
@@ -70,6 +95,17 @@ extern const struct emul __emul_list_end[];
 /* Use the devicetree node identifier as a unique name. */
 #define EMUL_REG_NAME(node_id) (_CONCAT(__emulreg_, node_id))
 
+/* Get a unique identifier based on the given _dev_node_id's reg property and
+ * the bus its on. Intended for use in other internal macros when declaring {bus}_emul
+ * structs in peripheral emulators.
+ */
+#define Z_EMUL_REG_BUS_IDENTIFIER(_dev_node_id) (_CONCAT(_CONCAT(__emulreg_, _dev_node_id), _bus))
+
+/* Conditionally places text based on what bus _dev_node_id is on. */
+#define Z_EMUL_BUS(_dev_node_id, _i2c, _espi, _spi)                                                \
+	COND_CODE_1(DT_ON_BUS(_dev_node_id, i2c), (_i2c),                                          \
+		    (COND_CODE_1(DT_ON_BUS(_dev_node_id, espi), (_espi),                           \
+				 (COND_CODE_1(DT_ON_BUS(_dev_node_id, spi), (_spi), (-EINVAL))))))
```

Comment:
Can this simplified as:
```
#define Z_EMUL_BUS(_dev_node_id)                                                \
	COND_CODE_1(DT_ON_BUS(_dev_node_id, i2c), (i2c),                                          \
		    (COND_CODE_1(DT_ON_BUS(_dev_node_id, espi), (espi),                           \
				 (COND_CODE_1(DT_ON_BUS(_dev_node_id, spi), (spi), (-EINVAL))))))
```

Reply:
Not just used as:
`Z_EMUL_BUS(node_id, i2c, espi, spi)`